tests/integration.js

```diff
@@ -24,26 +24,63 @@ function testPencil(browser) {
         .click('.tool[title ~= Crayon]')
         .assert.cssClassPresent('.tool[title ~= Crayon]', ['curTool'])
         .executeAsync(function (done) {
+            // utility function for returning a promise that resolves after a delay
+            // https://stackoverflow.com/a/6921279
+            function delay(t) {
+                return new Promise(function (resolve) {
+                    setTimeout(resolve, t);
+                });
+            }
+
+            Promise.delay = function (fn, t) {
+                // fn is an optional argument
+                if (!t) {
+                    t = fn;
+                    fn = function () {};
+                }
+                return delay(t).then(fn);
+            };
+
+            Promise.prototype.delay = function (fn, t) {
+                // return chained promise
+                return this.then(function () {
+                    return Promise.delay(fn, t);
+                });
+
+            };
+
             Tools.setColor('#123456');
             Tools.curTool.listeners.press(100, 200, new Event("mousedown"));
-            setTimeout(() => {
-                Tools.curTool.listeners.move(300, 400, new Event("mousemove"));
+            Promise.delay(() => {
+                Tools.curTool.listeners.release(300, 400, new Event("mouseup"));
+            }, 100)
+            .delay(() => {
+                Tools.setColor('#abcdef');
+                Tools.curTool.listeners.press(200, 100, new Event("mousedown"));
+            }, 100)
+            .delay(() => {
+                Tools.curTool.listeners.move(300, 200, new Event("mousemove"));
+            }, 100)
+            .delay(() => {
+                Tools.curTool.listeners.release(400, 100, new Event("mouseup"));
                 done();
             }, 100);
         })
-        .assert.visible("path[d='M 100 200 C 100 200 300 400 300 400'][stroke='#123456']")
+        .assert.visible("path[d='M 100 200 L 100 200 C 100 200 300 400 300 400'][stroke='#123456']")
+        .assert.visible("path[d='M 200 100 L 200 100 C 200 100 252.85954792089683 200 300 200 C 347.14045207910317 200 400 100 400 100'][stroke='#abcdef']")
         .refresh()
-        .assert.visible("path[d='M 100 200 C 100 200 300 400 300 400'][stroke='#123456']")
+        .assert.visible("path[d='M 100 200 L 100 200 C 100 200 300 400 300 400'][stroke='#123456']")
+        .assert.visible("path[d='M 200 100 L 200 100 C 200 100 252.85954792089683 200 300 200 C 347.14045207910317 200 400 100 400 100'][stroke='#abcdef']")
```

Comment:
Isn't there a rounding problem here, with `252.85954792089683` ?

Reply:
I haven't investigated how the code creates those extrapolated points for the bezier curves so I can't comment on that. Instinctively however I would have expected `250` and `350` instead of the current floating point values.